generator/src/main/java/com/scottlogic/deg/generator/Guice/ProfileProvider.java

```diff
@@ -21,10 +22,19 @@ public ProfileProvider(GenerationConfigSource configSource, ProfileReader profil
 
     @Override
     public Profile get() {
+        if (profile != null) {
+            return profile;
+        }
+
         try {
-            return profileReader.read(configSource.getProfileFile().toPath());
+            profile = profileReader.read(configSource.getProfileFile().toPath());
+            return profile;
         } catch (IOException | InvalidProfileException e) {
             throw new RuntimeException(e);
         }
     }
+
+    public void setProfile(Profile profile) {
```

Comment:
This appears to only exist for tests, see comment lower down about how it could be removed